### ISSUE TITLE
AKS 1.14 (preview) - unable to create new native thread

### ISSUE THREAD

**Issue by thebitguru** [Azure/AKS#1051]
**What happened**:
Java applications seem to have a problem creating native threads.  I realize that 1.14 is in preview and, in my case, this is not related to a production cluster.  I use protractor/selenium for automated testing and the same webdriver was working OK in 1.12, but upgrading to 1.14 preview causes this error.

```
[chrome #01-31] [20:38:21] I/hosted - Using the selenium server at http://selenium-hub:4444/wd/hub
[chrome #01-31] [20:38:21] E/driverProvider - Error code: 135
[chrome #01-31] [20:38:21] E/driverProvider - Error message: unable to create new native thread
[chrome #01-31] [20:38:21] E/driverProvider - Error: unable to create new native thread
[chrome #01-31]     at Hosted.<anonymous> (/data/node_modules/protractor/built/driverProviders/driverProvider.js:69:23)
[chrome #01-31]     at Generator.throw (<anonymous>)
[chrome #01-31]     at rejected (/data/node_modules/protractor/built/driverProviders/driverProvider.js:5:65)
[chrome #01-31]     at process._tickCallback (internal/process/next_tick.js:68:7)
```

I am seeing a few more reports of a [similar behavior with sonarqube](https://github.com/SonarSource/docker-sonarqube/issues/265).

**What you expected to happen**:
Work properly.

**How to reproduce it (as minimally and precisely as possible)**:
Sorry, my setup is fairly complicated because it requires angular and accompanying tests.  So, maybe try running a Java app, maybe SonarQube?

**Anything else we need to know?**:

**Environment**:
- Kubernetes version (use `kubectl version`):
```
Client Version: version.Info{Major:"1", Minor:"14", GitVersion:"v1.14.2", GitCommit:"66049e3b21efe110454d67df4fa62b08ea79a19b", GitTreeState:"clean", BuildDate:"2019-05-16T18:55:03Z", GoVersion:"go1.12.5", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"14", GitVersion:"v1.14.0", GitCommit:"641856db18352033a0d96dbc99153fa3b27298e5", GitTreeState:"clean", BuildDate:"2019-03-25T15:45:25Z", GoVersion:"go1.12.1", Compiler:"gc", Platform:"linux/amd64"}
```

- Size of cluster (how many worker nodes are in the cluster?)
5

- General description of workloads in the cluster (e.g. HTTP microservices, Java app, Ruby on Rails, machine learning, etc.)
One Selenium hub, 5 Chrome Nodes, one nginx instance, and this protractor job.

**Comment by joshua-alday** [comment]
I get similar issues with jboss/wildfly and maven builds in certain situations.

**Comment by joshua-alday** [comment]
I just want to confirm that I'm seeing this problem as well with more details.

**What happened:**
When using kubernetes 1.14, executing JVM based applications will randomly result in OOM: Unable to create native thread. Running maven builds will sometimes run fine, but running wildfly will always result in this issue. Monitoring the applications, there seems to be no major uptick in memory consumption and cpu seems to work properly with set cgroupings.

(One exception is running jvm applications or maven builds manually within the pod, though this could be due to the lack of jenkins agent jar adding to the complexity of execution.)

**What you expected to happen:**
The JVM application runs with no OOM issues aside from those induced by malformed user settings and configs (basically human error should only be the playing factor here.)

**How to reproduce it (as minimally and precisely as possible):**
Set up AKS with kubernetes version 1.14. 
Set any size single node (I've tried different standard sizes, compute and memory optimized.)
Run a JVM based application like wildfly or jenkins maven builds.

**Anything else we need to know?:**
Downgrading to 1.12 seems to have resolved this issue for me. Although this can definitely work for the interim, I require features from 1.14 in the near future.

**Comment by monaka** [comment]
As I commented to [the another issue](https://github.com/SonarSource/docker-sonarqube/issues/265#issuecomment-509124540), this will be cause by the too small value to `pod-max-pids`.
I think it's better to increase it more than 100 (current) because each native thread will be counted as pid.

**Comment by palma21** [comment]
@monaka is correct.
And we are increasing that value, expect it to be fixed as we bring 1.14 to GA on AKS.

**Comment by palma21** [comment]
Fixed with (and before) the GA of 1.14 as noted in
https://github.com/Azure/AKS/releases/tag/2019-07-29
